Homework 2

alenasidorova1111 opened the pull request:
Homework 2 completed

alenasidorova1111 committed:
Added:
.gitignore

alenasidorova1111 committed:
Added:
DataProviders.java
EPAMTestApp.apk
README.md

Updated:
BaseTest.java
IDriver.java
IPageObject.java
NativeMobileTests.java
NativePageObject.java
nativeTNG.xml
PageObject.java
pom.xml
WebMobileTests.java
WebPageObject.java
webTNG.xml

alenasidorova1111 committed:
Updated:
BaseTest.java
WebMobileTests.java

MaksimMeshcheriakovEPAM commented on src/test/java/scenarios/WebMobileTests.java:
```diff
@@ -0,0 +1,33 @@
+
+        // Check that 1st search result contains searchPhrase
+        getPage().getWebElement("searchField").sendKeys(searchPhrase, Keys.ENTER);
+        assert (getPage().getWebElements("searchResults").get(1).getText().contains(searchPhrase));
```
А почему вы на релевантность только 1й эл-т выдачи проверяете? А как же остальные?

MaksimMeshcheriakovEPAM commented on src/test/java/scenarios/WebMobileTests.java:
```diff
@@ -0,0 +1,33 @@
+        assert ((WebDriver) getDriver()).getTitle().equals(pageTitle) : "This is not Google page";
+
+        // Set opened page
+        setPageObject(appType, getDriver());
```
А это здесь зачем? PageObject же уже установлен в процессе инициализации сьюта (@BeforeSuite)?

alenasidorova1111 replied:
Был NullPointer, переписала решение на рассмотренное в последней лекции)

MaksimMeshcheriakovEPAM commented on src/main/java/setup/DataProviders.java:
```diff
@@ -0,0 +1,31 @@
+
+    @DataProvider(name = "dataProviderForNativeTest")
+    public static Object[][] dataProviderForNativeTest() {
+        String email = "sid@gmail.com";
```
Вынесение присвоения значений в отдельный класс по сути ничего не поменяло - все равно значения в коде. Т.е. при их изменении (процесс поддержки тестов) придется перекомпилировать код. Best practice - зачитывать значения в код из внешних текстовых файлов

MaksimMeshcheriakovEPAM commented on src/main/java/pageObjects/NativePageObject.java:
```diff
@@ -6,14 +6,36 @@
+    WebElement registerNewAccBtn;
+    @AndroidFindBy(id = "platkovsky.alexey.epamtestapp:id/registration_cancel_button")
+    WebElement cancelBtn;
+    @AndroidFindBy(xpath = "/hierarchy/android.widget.FrameLayout/android.widget.LinearLayout/android.widget." +
```
Не очень хорошо испоьзовать абс. xpath - хрупкие. Лучше относительные, например по тексту в искомом элементе.

alenasidorova1111 replied:
Передела, но возник вопрос: зачем тогда ассер по тексту элемента? Если мы его и ищем по тексту... Получается ассерт не нужен, т.к. если элемент будет найден по тексту, то смысл проверять этот текст нет....

alenasidorova1111 replied:
Ну и текст тоже может поменяться, к примеру тот же используемый нами заголовок...

MaksimMeshcheriakovEPAM reviewed (changes requested):
Два серьезных прокола (см. комменты в коде ПР):
1) Не до конца корректная проверка web mobile
2) Фактически присвоение значений переменных осталось в коде, только вынесено в отдельный класс

alenasidorova1111 committed:
Updated:
BaseTest.java
DataProviders.java
NativePageObject.java
pom.xml
WebMobileTests.java

Added:
DataForNativeAndroidTests.json
DataForWebTests.json
NativeAndroidTestData.java
WebTestData.java

alenasidorova1111 committed:
Updated:
DataProviders.java

Reason: testData directory deleted, its files were moved to src/test/resources
